File: src/main/java/com/temelyan/pomoapp/web/project/ProjectRestController.java
package com.temelyan.pomoapp.web.project;

import com.fasterxml.jackson.annotation.JsonView;
import com.temelyan.pomoapp.JsonViews.ProjectViews;
import com.temelyan.pomoapp.model.Project;
import org.springframework.http.MediaType;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.Set;

@RestController
@RequestMapping("/ajax/project/")
public class ProjectRestController extends AbstractProjectController {

    @PostMapping(path = "add")
    public void addProject(String name) {
        Project project = new Project(name);
        super.create(project);
    }

    @JsonView(ProjectViews.IncludeTasks.class)
    @GetMapping(path = "get", produces = MediaType.APPLICATION_JSON_VALUE)
    public Set<Project> getAllProjects() {
        return super.getAll();
    }
}

File: src/main/java/com/temelyan/pomoapp/web/project/ProjectRestController.kt
package com.temelyan.pomoapp.web.project

import com.fasterxml.jackson.annotation.JsonView
import com.temelyan.pomoapp.JsonViews.ProjectViews
import com.temelyan.pomoapp.model.Project
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/project/")
class ProjectRestController : AbstractProjectController() {

    @JsonView(ProjectViews.IncludeTasks::class)
    @GetMapping("get")
    fun getAllProjects(): Set<Project> {
        return super.all
    }

    @PostMapping(path = ["add"])
    fun addProject(name: String) {
        val project = Project(name)
        super.create(project)
    }
}
